[DOCS] Correct `source.index` parm def in reindex API

jrodewig opened the pull request:
Corrects the `source.index` request body parameter definition to
state that arrays, not comma-separated values, are accepted.

Relates to #51949

elasticmachine commented:
Pinging @elastic/es-docs (>docs)

elasticmachine commented:
Pinging @elastic/es-distributed (:Distributed/Reindex)

henningandersen commented:
I think we should just support the "," notation instead to try to align with the [multiple indices documentation](https://www.elastic.co/guide/en/elasticsearch/reference/7.5/multi-index.html). This is also how snapshot/restore works.

henningandersen commented:
Opened #52044 to fix this.

jrodewig commented:
Thanks @henningandersen. However, I think this would still be a relevant docs update for 7.6 and 7.5.

I've changed the target branch and updated the labels appropriately.

jrodewig commented:
Thanks @henningandersen. After taking a closer look at the 6.8 docs, I believe this is already documented pretty well:

<img width="764" alt="Screen Shot 2020-02-19 at 7 34 05 AM" src="https://user-images.githubusercontent.com/40268737/74834862-40ba5680-52ea-11ea-935f-222e4812f258.png">

I'll close this PR. Thanks for fixing this!
